Shooter

ThaiPituk opened the pull request:
Added Shooter.h which controls velocity at any input.

ThaiPituk committed:
Shooter

Added Shooter.h which controls velocity at any input.

jjxfan commented:
Few comments before you change this on Thursday. 
1. Look at our motor naming conventions which should be linked on the resources docs and look into using settings.h to predefine button maps.
2. From what I'm seeing this code works like this: Hold button and pneumatic goes down, and intake motor intakes balls, release the button and the intake goes up and it stops the intake. If motor amps go above 10, raise intake and spit out balls. Is this intended and have you looked into using a toggle? Reference Bryan's toggle motor code if you need help.

ThaiPituk committed:
Shooter 2

Added motor lead and follow, changed motor names, added invert.

rwang5412 committed:
Shooter3

We added two control types, PercentOutput controls and the ability to control the motors independently.

rwang5412 committed:
Merge branch 'dev-shooter' of https://github.com/FRC-Team-955/955-2022-RobotCode into dev-shooter

owen-mcmanus committed:
revmoved extra lines

owen-mcmanus commented on src/main/cpp/shooter.cpp:
```diff
@@ -0,0 +1,23 @@
+using namespace std; 
+using namespace frc; 
+
+float Shooter::ShootAtVelocity(int Velocity) {
```
lowecase v

owen-mcmanus commented on src/main/cpp/shooter.cpp:
```diff
@@ -0,0 +1,23 @@
+}
+
+void Shooter::ShootPercentOutput(int Percent){
+    shootertalon_lead.Set(ControlMode::PercentOutput, Percent);
```
set other motor to follower

owen-mcmanus replied:
and lowercase p

owen-mcmanus commented on src/main/cpp/shooter.cpp:
```diff
@@ -0,0 +1,23 @@
+    shootertalon_lead.Set(ControlMode::PercentOutput, Percent);
+}
+
+float Shooter::IndependentControl(int VelocityLead, int VelocityFollow) {
```
lowercase and snake_case

owen-mcmanus commented on src/main/include/shooter.h:
```diff
@@ -0,0 +1,22 @@
+#include "frc/Joystick.h"
+#include "ctre/Phoenix.h"
+using namespace std;
+using namespace frc;
```
don't use std if you dont need it

owen-mcmanus commented on src/main/include/shooter.h:
```diff
@@ -0,0 +1,22 @@
+using namespace frc;
+
+class Shooter {
+    Shooter(){shootertalon_follow.SetInverted(1);}
```
make a const in settings.h to use pass into `SetInverted()`

ThaiPituk committed:
Shooter 4

Fixed formatting, added constant in settings.h.

rwang5412 committed:
Shooter3

We added two control types, PercentOutput controls and the ability to control the motors independently.

rwang5412 committed:
Merge branch 'dev-shooter' of https://github.com/FRC-Team-955/955-2022-RobotCode into dev-shooter

owen-mcmanus committed:
formatting, naming, and settings.h

owen-mcmanus committed:
Merge branch 'beta' into dev-shooter
